internal/coreinternal/aggregateutil/aggregate.go

```diff
@@ -296,17 +296,31 @@ func mergeExponentialHistogramDataPoints(dpsMap map[string]pmetric.ExponentialHi
 			}
 			dp.SetCount(dp.Count() + dps.At(i).Count())
 			dp.SetSum(dp.Sum() + dps.At(i).Sum())
+			dp.SetZeroCount(dp.ZeroCount() + dps.At(i).ZeroCount())
 			if dp.HasMin() && dp.Min() > dps.At(i).Min() {
 				dp.SetMin(dps.At(i).Min())
 			}
 			if dp.HasMax() && dp.Max() < dps.At(i).Max() {
 				dp.SetMax(dps.At(i).Max())
 			}
+			// Merge bucket counts.
+			// Note that groupExponentialHistogramDataPoints() has already ensured that we only try
+			// to merge exponential histograms with matching Scale and Positive/Negative Offsets,
+			// so the corresponding array items in BucketCounts have the same bucket boundaries.
+			// However, the number of buckets may differ depending on what values have been observed.
 			for b := 0; b < dps.At(i).Negative().BucketCounts().Len(); b++ {
```

Comment:
I see how this should handle cases where negatives.Len() > dps.At(i).Negatives.BucketCounts().Len(), but I think it should be covered in tests to ensure that future maintenance doesn't break the behavior accidentally.